src/routers/user.ts

```diff
@@ -1,24 +1,51 @@
 import express from 'express';
+import { create, read, update, del } from '../operators/UserOperations';
 
 const userRouter = express.Router();
+const userError = 'User not found.';
 
-userRouter.get('/', (req, res) => {
-	console.log('Get User');
-	res.send('Get userRouter');
+userRouter.get('/:id', async (req, res) => {
+	try {
+		console.log('Get User');
+		const user = await read(req.params.id);
+		res.send({ ...user, semesters: Array.from(user?.semesters || []) });
```

Comment:
The res.send here can fail, if semesters is null, `Array.from(null)` throws an error. We should do something like

```ts
const user = await read(req.params.id);
const semesters = user?.semesters ? Array.from(user.semesters) : [];
res.send({ ...user, semesters});
```

Reply:
Nvm read the code wrong!